src/index.spec.js

```diff
@@ -50,6 +50,32 @@ describe('provideState', () => {
     })
   })
 
+  describe('resetState', () => {
+    it('is called to reset the state to its initial values', async () => {
+      const props = { bar: 'baz' }
+      const { effects, getInjectedState } = makeTestInstance(
+        {
+          initialState: () => ({
+            foo: 'bar',
+          }),
+          effects: {
+            changeState (_, value) {
+              return { foo: value }
+            },
+            async reset () {
+              await this.resetState()
+            },
+          },
+        },
+        props
+      )
+      await effects.changeState('foo')
+      expect(getInjectedState()).toEqual({ foo: 'foo' })
+      await effects.reset()
+      expect(getInjectedState()).toEqual({ foo: 'bar' })
+    })
+  })
```

Comment:
While you're at it, can you also test that `resetState` is passed to children via `injectState`?